src/github.com/getlantern/flashlight/logging/logging.go

```diff
@@ -67,108 +54,132 @@ func Setup(version string, buildDate string) *rotator.SizeRotator {
+	errorOut = timestamped(io.MultiWriter(os.Stderr, logFile))
+	debugOut = timestamped(io.MultiWriter(os.Stdout, logFile))
+	golog.SetOutputs(errorOut, debugOut)
+}
 
+func Configure(cfg *config.Config, version string, buildDate string) {
 	if logglyToken == "" {
 		log.Debugf("No logglyToken, not sending error logs to Loggly")
-	} else {
-		log.Debugf("Sending error logs to Loggly")
-		logglyClient = &logglyErrorWriter{loggly.New(logglyToken)}
-		errorOut = NonStopWriter(errorOut, logglyClient)
+		return
 	}
 
-	debugOut := timestamped(NonStopWriter(os.Stdout, file))
-	golog.SetOutputs(errorOut, debugOut)
-	return file
-}
+	if version == "" {
+		log.Error("No version configured, Loggly won't include version information")
+		return
+	}
 
-func Configure(cfg *config.Config) (err error) {
-	cfgMutex.Lock()
-	defer cfgMutex.Unlock()
+	if buildDate == "" {
+		log.Error("No build date configured, Loggly won't include build date information")
+		return
+	}
 
-	if logglyClient == nil {
-		return fmt.Errorf("No loggly client to configure.")
+	cfgMutex.Lock()
+	if cfg.Addr == lastAddr {
+		cfgMutex.Unlock()
+		log.Debug("Logging configuration unchanged")
+		return
 	}
 
+	// Using a goroutine because we'll be using waitforserver and at this time
+	// the proxy is not yet ready.
+	go func() {
+		lastAddr = cfg.Addr
+		enableLoggly(cfg, version, buildDate)
+		cfgMutex.Unlock()
+	}()
+}
+
+func Close() error {
+	return logFile.Close()
+}
+
+// timestamped adds a timestamp to the beginning of log lines
+func timestamped(orig io.Writer) io.Writer {
+	return wfilter.LinePrepender(orig, func(w io.Writer) (int, error) {
+		return fmt.Fprintf(w, "%s - ", time.Now().In(time.UTC).Format(logTimestampFormat))
+	})
+}
+
+func enableLoggly(cfg *config.Config, version string, buildDate string) {
 	if cfg.Addr == "" {
-		return fmt.Errorf("No proxy address provided.")
+		log.Error("No known proxy, won't report to Loggly")
+		removeLoggly()
+		return
 	}
 
-	if err = waitforserver.WaitForServer("tcp", cfg.Addr, 10*time.Second); err != nil {
-		return fmt.Errorf("Proxy never came online at %s: %q", cfg.Addr, err)
+	err := waitforserver.WaitForServer("tcp", cfg.Addr, 10*time.Second)
+	if err != nil {
+		log.Errorf("Proxy never came online at %v, not logging to Loggly", cfg.Addr)
+		removeLoggly()
+		return
 	}
 
-	var cli *http.Client
-	if cli, err = util.HTTPClient(cfg.CloudConfigCA, cfg.Addr); err != nil {
-		return fmt.Errorf("Could not create proxy HTTP client %q.", err)
+	var client *http.Client
+	client, err = util.HTTPClient(cfg.CloudConfigCA, cfg.Addr)
+	if err != nil {
+		log.Errorf("Could not create proxied HTTP client, not logging to Loggly: %v", err)
+		removeLoggly()
+		return
 	}
 
-	logglyClient.l.SetHTTPClient(cli)
+	log.Debugf("Sending error logs to Loggly via proxy at %v", cfg.Addr)
 
-	return nil
+	lang, _ := jibber_jabber.DetectLanguage()
+	logglyWriter := &logglyErrorWriter{
+		lang:            lang,
+		tz:              time.Now().Format("MST"),
```

Comment:
Great question.  We actually use the timezone to identify the user's location in case geolocation doesn't work, so we actually want the user's timezone, not the reported time.

When you look in Loggly, you can see that the timestamp is displayed with its timezone, so there should be little confusion.

I am going to rename timeZone to userTimeZone just to make it clearer.

![screen shot 2015-02-25 at 11 11 12 am](https://cloud.githubusercontent.com/assets/5000654/6375793/135a6bf4-bcdf-11e4-9b21-4e129263087a.png)